### PR TITLE
Refactor TestStatus

### PR DESCRIPTION
**Pull request by jgfouca**
Make recent PENDing state change work regardless of whether
a phase was run from create_test or by hand.

Other more minor refactors like demoting the INIT_PHASE from an
official phase to just a book-keeping item for test_scheduler.

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: Yes, changes in management of TestStatus file

Code review: jedwards, billsacks

### CODE REVIEW

**Comment by jgfouca** [comment]
Put up for review. Tests still running; I'll post here when they're done.

File: utils/python/CIME/test_status.py [hunk 1]
```diff
@@ -101,25 +107,72 @@ def get_name(self):
 
     def set_status(self, phase, status, comments=""):
         """
-        >>> with TestStatus(test_dir="/", test_name="ERS.foo.A") as ts:
+        Update the status of this test by changing the status of given phase to the
```
Comment (jgfouca) [inline utils/python/CIME/test_status.py]: Key changes are all here.

**Comment by billsacks** [comment]
Thanks for this. I'm interested to look at it, but I am so swamped with things right now that I'm not sure when I'll get a chance - especially given that I'm taking much of Friday off. So if Jim E approves this and you're eager to get it to master, you don't need to wait on my behalf... in that case, I'll try to take a look at some point once it's on master.

**Comment by billsacks** [comment]
@jgfouca - I finally got a chance to look at this PR. Thank you very much for doing this! I like how you implemented it. It seems solid, giving me the confidence I was looking for. I especially appreciate how you handled the rerunning of a phase.

I have just a couple of minor cleanup things (documentation, tweaking a unit test)... I'll submit them in an upcoming PR.

**Pull request merged**